src/backend/renderer/element/mod.rs

```diff
@@ -317,6 +317,19 @@ impl RenderElementStates {
     }
 }
 
+/// Defines the kind of an [`Element`]
+///
+/// This can give the backend a hint about how
+/// to handle the element
```

Comment:
It would be good to document here what exactly this affects (cursor plane assignment) and why it matters (animating elements will cause performance problems). Because right now you might assume that it's a good idea to set the DnD surface's kind to cursor, but the DnD surface can very well animate. For example, in one of my apps the DnD surface can show a playing video preview:

https://github.com/Smithay/smithay/assets/1794388/f77298b9-f4a4-4bcd-bdbf-be6865c384d2

Also libadwaita's tab DnD surface will animate its size occasionally.

Reply:
Good point, I will clarify what this `Kind` is intended for. I want to keep it generically and not depend on the `DrmCompositor` internals, so would something like:
> An element representing a cursor is expected to change infrequently. Failing to mark cursor elements can result in sub-optimal performance. In contrast, marking elements that change frequently as `Cursor` can degrade performance significantly. `

Reply:
Sounds good, perhaps only reword this a little bit:

> Failing to mark cursor elements can result in sub-optimal performance.

Maybe

> Not marking a cursor element as `Cursor` may result in lower performance and increased power usage.

Reply:
done